api/src/main/java/org/ccci/idm/user/util/HashUtility.java

```diff
@@ -0,0 +1,21 @@
+package org.ccci.idm.user.util;
+
+import org.springframework.security.crypto.bcrypt.BCrypt;
+
+/**
+ * Utility class encapsulating a particular hashing algorithm.
+ *
+ * author@lee.braddock
+ */
+public class HashUtility {
+
+    private static final Integer BCRYPT_WORK_FACTOR = 12;
+
+    public static String getHash(final String string) {
+        return BCrypt.hashpw(string, BCrypt.gensalt(BCRYPT_WORK_FACTOR));
+    }
+
+    public static boolean checkHash(final String string, final String hash) {
+        return BCrypt.checkpw(string, hash);
+    }
```

Comment:
I would add in a simple string normalization function that strips leading & trailing whitespace and converts to lowercase to help ease SA matching.